src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -473,6 +550,19 @@ public RestoreSnapshotRequest source(Map source) {
                 } else {
                     throw new ElasticsearchIllegalArgumentException("malformed rename_replacement");
                 }
+            } else if (name.equals("index_settings")) {
+                if (!(entry.getValue() instanceof Map)) {
+                    throw new ElasticsearchIllegalArgumentException("malformed index_settings section, should indices an inner object");
+                }
+                indexSettings((Map<String, Object>) entry.getValue());
+            } else if (name.equals("ignore_index_settings")) {
+                    if (entry.getValue() instanceof String) {
+                        ignoreIndexSettings(Strings.splitStringByCommaToArray((String) entry.getValue()));
+                    } else if (entry.getValue() instanceof ArrayList) {
```

Comment:
I think this should be `List` to allow more than just `ArrayList`s in the parameter